File: BinarySearch.py

#Given n numbers and a number k, write a program to search the no. k in the list using Binary search algo.
n=int(input("how maany numbers?"))
l1=[]
for i in range(0,n):
  a=int(input("enter number: "))
  l1.append(a)
l1.sort()
k=int(input("enter element to be searched? "))
beg=0
end=len(l1)-1
flag=0
while(flag==0):
  mid=(beg+end)//2
  if(k==l1[mid]):
    flag=1
    break;
  elif(k>l1[mid]):
    beg=mid+1
  elif(k<l1[mid]):
    end=mid-1
print("element found @",mid,"position")

File: BinarySearch.cpp
// Given n numbers and a number k, write a program to search the number. k in the list using Binary Search algorithm.

#include<iostream.h>
void main()
{
  int a[20],k,mid=0,f=0,i,n;
  cin>>n>>k;
  for(i=0;i<n;i++)
    cin>>a[i];
  int l=0,r=n-1;
  while(l<=r)
  {
    mid=(l+r)/2;
    if(k==a[mid])
    {f=1;
     break;
    }
    if(a[mid]<k)
    {
      l=mid+1;
    }
    if(a[mid]>k)
      r=mid-1;
  }
  if(f==0)
    cout<<"No. not found";
  else
    cout<<"found";
}

